### ISSUE TITLE
Animation longs last on iOS only (there is a delay on the end, compared to android)

### ISSUE THREAD

**Issue by vitosorriso** [IjzerenHein/react-native-shared-element#28]
Hello everyone, it's me, again! i'm trying to replicate the app-of-the-day Apple animation, using `react-navigation-shared-element`, there is just one last thing that I miss.

On iOS, compared to android, animation has the same length but ends with a delay:
(iPhone X on the left - iOS 13.3.1 , Xiaomi Mi 9 SE on the right - MIUI 11 based on Android 9)

![sp_animation_delay_ios](https://user-images.githubusercontent.com/19435474/78396888-6e383680-75f0-11ea-8b50-979041816a6f.gif)

As you can see, on iOS animations longs last compared to android (you can see from the X icon which clearly appears later on iOS); same thing when going back (you can see it from the borderRadius applied to the whole card, but maybe it's not too visible in the gif)

I tried to disable shadows (because it's the only different thing in my react-native code between android and ios), but nothing changed.

The side-effect is not just visual-only: during the animation, i disable every kind of interaction on the device, because these interactions could break the shared element transition (like in this gif where I disabled this mechanism):

![sp_animation_scrollenabled](https://user-images.githubusercontent.com/19435474/78398120-cec87300-75f2-11ea-99e6-06abea000d16.gif)

To do this, I just listen to `didFocus` react-navigation event, because I noticed that when this event is fired, I'm 100% sure animation is ended and I will not obtain glitches (same mechanism applied on both List and Details page):

```js
  useEffect(() => {
    // didFocus is called right after the transition ends!
    const onFocus = (): void => {
      setScrollEnabled(true)
    }

    const didFocusListener = navigation.addListener('didFocus', onFocus)

    return (): void => {
      didFocusListener.remove()
    }
  }, [])
```

So basically, all of this means: iOS user need to wait longer before interacting, compared to android.

Why is this happening? Ideas / suggestions on how to investigate on this?? Thanks in advance!

my packages (i changed react-native-shared-element to the alpha version due to that borderRadius bug, same behaviour happens with stable 0.5.6):
```
"react": "16.9.0",
"react-native": "0.61.5",
"react-native-shared-element": "0.7.0-alpha0", // "^0.5.6" same behaviour
"react-navigation": "^4.1.0",
"react-navigation-shared-element": "^2.2.0",
"react-navigation-stack": "^2.1.0",
"react-navigation-tabs": "^2.7.0"
```

**Comment by mrousavy** [comment]
Hi @vitosorriso ! Try using the following transition spec:

```js
const iosTransitionSpec = {
	animation: 'spring',
	config: {
		stiffness: 1000,
		damping: 500,
		mass: 3,
		overshootClamping: true,
		restDisplacementThreshold: 10,
		restSpeedThreshold: 10,
	},
};
```

..in your Shared Element Navigator like this:

```js
<SharedElementStack.Navigator
	mode="modal"
	screenOptions={{
		useNativeDriver: true,
		gestureEnabled: false,
                ...TransitionPresets.ModalSlideFromBottomIOS,
		transitionSpec: {
			open: iosTransitionSpec,
			close: iosTransitionSpec,
		},
```

Also - How did you manage to implement the slide down gesture to go back? When I use `gestureEnabled: true` it just uses the `cardStyleInterpolator` I created (fading in and out using opacity property) but no Shared Element transition..

**Comment by IjzerenHein** [comment]
Hi! This problem is not related to the native implementation, but rather due to the fact that the react-navigation spring animation on iOS takes a long time to settle. I'm moving this to the react-navigation-shared-element repo.
